spring/src/main/java/io/cloudevents/spring/http/CloudEventHttpUtils.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2019-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cloudevents.spring.http;
+
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import io.cloudevents.CloudEventContext;
+import io.cloudevents.core.CloudEventUtils;
+import io.cloudevents.core.builder.CloudEventBuilder;
+import io.cloudevents.core.message.MessageReader;
+import io.cloudevents.http.HttpMessageFactory;
+import io.cloudevents.http.impl.HttpMessageWriter;
+
+import org.springframework.http.HttpHeaders;
+
+/**
+ * Miscellaneous utility methods to assist with Cloud Events in the context of Spring Web
+ * frameworks. Primarily intended for the internal use within Spring-based frameworks or
+ * integrations.
+ *
+ * @author Dave Syer
+ * @since 2.0
+ */
+public class CloudEventHttpUtils {
+
+	public static MessageReader toReader(HttpHeaders headers, Supplier<byte[]> body) {
+		return HttpMessageFactory.createReaderFromMultimap(headers, body.get());
+	}
+
+	public static HttpMessageWriter toWriter(HttpHeaders headers, Consumer<byte[]> sendBody) {
+		return HttpMessageFactory.createWriter((k, v) -> headers.set(k, v), sendBody);
+	}
+
+	public static HttpHeaders toHttp(CloudEventContext event) {
+		HttpHeaders headers = new HttpHeaders();
+		CloudEventUtils.toReader(CloudEventBuilder.fromContext(event).build()).read(toWriter(headers, bytes -> {
+		}));
+		return headers;
+	}
+
+	public static CloudEventBuilder fromHttp(HttpHeaders headers) {
+		return CloudEventBuilder
+				.fromContext(CloudEventUtils.toEvent(CloudEventHttpUtils.toReader(headers, () -> null)));
+	}
```

Comment:
Since these 2 are used only in tests now, can we keep them inside tests now? If we need them in production code, we can move them later

Reply:
The tests are more like integration tests illustrating example usage. If we add actual example apps I would add the same code there, so no, I don't think these can be private or test scoped.

Reply:
I think this is related to this point here: https://github.com/cloudevents/sdk-java/pull/312#discussion_r531494254

Reply:
Yes. So as long as we need to support the user doing conversion from `CloudEventContext` to `HttpHeaders` then we need this utility to be public.